### PR TITLE
Add a --benchmark mode to hot run

### PR DESCRIPTION
**Pull request by johnmccutchan**
@Hixie @sethladd

### CODE REVIEW

File: packages/flutter_tools/lib/src/hot.dart [hunk 1]
```diff
@@ -132,6 +133,8 @@ class HotRunner extends ResidentRunner {
   String _projectRootPath;
   Set<String> _startupDependencies;
   final AssetBundle bundle = new AssetBundle();
+  final bool benchmarkMode;
```
Comment (sethladd) [inline packages/flutter_tools/lib/src/hot.dart]: do we want to default this to an explicit value?

Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/hot.dart]: The constructor argument has a default value

**Comment by Hixie** [comment]
Can we run this benchmark by default on travis? (in dev/bots/test.sh)

**Comment by Hixie** [comment]
(mostly to verify that hot reload works)

**Comment by johnmccutchan** [comment]
@Hixie The plan is to run it as part of @yjbanov's benchmark suite.

**Comment by dragostis** [comment]
@johnmccutchan I think it would be a good idea to have a small test on Travis too just to make sure we don't break hot reload before merging.

**Comment by johnmccutchan** [comment]
@dragostis Does travis run with a mobile device attached?

**Comment by Hixie** [comment]
LGTM

**Comment by dragostis** [comment]
@johnmccutchan Good point! I guess the purpose was not to overburden the PR, but more of an open question on how to make sure we don't brake runtimes? (we've had a few issues lately with `--profile` `--release` lately which no regression tests captured)

**Pull request merged**